### PR TITLE
Document Organizations API

### LINKED ISSUE #410: Document Organizations API
We should document this endpoint using a combination of:
- [API Blueprint](https://apiblueprint.org/) for writing the documentation of resources, actions, and URIs
- [Dredd](https://dredd.readthedocs.io/en/latest/) for testing the documentation matches the API
- [Apiary](https://apiary.io/) for browsing the docs

Please reference the journey issue #387 for further instructions or commentary on documenting.

### PR DESCRIPTION
**Pull request by npendery**
This update adds API documentation for Organizations. The following actions have been documented:
- Create an organization – `POST /organizations`
- Find an organization – `GET /organizations/{id}`
- Update an organization – `PATCH /organizations/{id}`
- Filter organizations - `GET /organizations`

Closes #410

### CODE REVIEW

File: blueprint/api.apib [hunk 1]
```diff
@@ -150,6 +150,112 @@ Exchange credentials for `access_token`.
...
+
+  + Attributes (Forbidden Response)
+
++ Response 422 (application/vnd.api+json)
+
+  + Attributes (Unprocessable Entity Response)
+
+## Organization [/organizations/{id}]
+
+### Retrieve an organization [GET]
```
Comment (joshsmith) [inline blueprint/api.apib:206]: I would remove the duplication of `Organization` and place the URL with the method, like `[GET /organizations/{id}]`. We should update style guide for this.

File: blueprint/api.apib [hunk 2]
```diff
@@ -150,6 +150,112 @@ Exchange credentials for `access_token`.
...
+  + Attributes (Organization Response)
+
++ Response 404 (application/vnd.api+json)
+
+  + Attributes (Record Not Found Response)
+
+## Organization [/organizations/{id}]
+
+### Update an organization [PATCH]
+
```
Comment (joshsmith) [inline blueprint/api.apib:227]: Same here.

Comment (joshsmith) [inline blueprint/api.apib:227]: Also, is it `PUT` or `PATCH`. We use both interchangeably and it's not clear what we _should_ be using.

File: blueprint/api.apib [hunk 3]
```diff
@@ -150,6 +150,112 @@ Exchange credentials for `access_token`.
...
+    + Attributes (Forbidden Response)
+
++ Response 422 (application/vnd.api+json)
+
+    + Attributes (Unprocessable Entity Response)
+
+## Organizations [/organizations]
+
+### Filters organizations by id [GET]
```
Comment (joshsmith) [inline blueprint/api.apib]: Same here with header.

File: blueprint/api.apib [hunk 4]
```diff
@@ -150,6 +150,112 @@ Exchange credentials for `access_token`.
...
+
++ Parameters
+
+  + filter: `1` (enum[number], required) - Id query to filter organizations
```
Comment (joshsmith) [inline blueprint/api.apib]: The `filter` here is not exactly clear that you can accept multiple `id`s.

File: blueprint/api.apib [hunk 5]
```diff
@@ -527,14 +633,39 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
...
++ relationships
+  + members
+    + data(array[Members Resource Identifier])
+    + links
+  + projects
+    + data(array[Projects Resource Identifier])
+    + links
+  + organization_memberships
+    + data(array[Organization Memberships Resource Identifier])
+    + links
```
Comment (joshsmith) [inline blueprint/api.apib]: Are we indenting four or two spaces? @christopherstyles I think you came down somewhere on this one.

File: blueprint/api.apib [hunk 6]
```diff
@@ -527,14 +633,39 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
...
++ data(array[Organization Resource])
+
+## Organization Resource (object)
++ include Organization Resource Identifier
++ attributes
+  + name: `Code Corps`
+  + slug: `code_corps`
+  + description: `Build a better future.`
+  + icon_thumb_url: `https://random_string.cloundfront.net/organizations/1/thumb.png`
+  + icon_large_url: `https://random_string.cloundfront.net/organizations/1/large.png`
```
Comment (joshsmith) [inline blueprint/api.apib]: Do we want to use `random_string` here or something more explicit.

File: blueprint/api.apib [hunk 7]
```diff
@@ -527,14 +633,39 @@ This allows A Skill to be under different Roles and a Role to have multiple Skil
...
++ data(Organization Resource)
+
+## Organizations Response (object)
++ data(array[Organization Resource])
+
+## Organization Resource (object)
++ include Organization Resource Identifier
++ attributes
+  + name: `Code Corps`
+  + slug: `code_corps`
```
Comment (joshsmith) [inline blueprint/api.apib]: I believe by default this slug is dasherized and not underscored. Could be wrong.

File: blueprint/api.apib (hunk 2 above)
Comment (christopherstyles) [inline blueprint/api.apib:227]: It should be `PATCH`.

File: blueprint/api.apib [hunk 8]
```diff
@@ -150,6 +150,112 @@ Exchange credentials for `access_token`.
...
+
+      Accept: application/vnd.api+json
+
++ Response 200 (application/vnd.api+json)
+
+  + Attributes (Organization Response)
+
++ Response 404 (application/vnd.api+json)
+
+  + Attributes (Record Not Found Response)
```
Comment (joshsmith) [inline blueprint/api.apib]: Looks like inconsistent indentation in the Attributes.

File: blueprint/api.apib [hunk 9]
```diff
@@ -150,6 +150,112 @@ Exchange credentials for `access_token`.
 
 # Group Organizations
 
+This endpoint retrieves organzations that have projects
```
Comment (joshsmith) [inline blueprint/api.apib]: I think this description could use some love. What are organizations in the context of Code Corps? Would also add specific descriptions to each endpoint, indicating things like:
- who can create an organization?

**Comment by joshsmith** [comment]
Overall looks really good. 👍  I would make some of those minor changes suggested here and then we should be good to go. Maybe resubmit for another review at that point?

File: blueprint/api.apib (hunk 7 above)
Comment (npendery) [inline blueprint/api.apib]: I saw in the `db/fixures` file for organizations it was underscored so went with that format

File: blueprint/api.apib (hunk 6 above)
Comment (npendery) [inline blueprint/api.apib]: Not sure. It is pretty much just a string of random characters so thats why I initially put this. Do you think we should make it a realistic cloudfront address?

File: blueprint/api.apib (hunk 1 above)
Comment (npendery) [inline blueprint/api.apib:206]: Going to change it to how they have it in [the examples](https://github.com/apiaryio/api-blueprint/blob/master/examples/10.%20Data%20Structures.md), think that makes sense

**Comment by joshsmith** [comment]
👍

**Comment by joshsmith** [comment]
Merge at will.

**Pull request merged**